qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -367,9 +369,77 @@ def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_acti
     return spinz_matrix_elements
 
 
+def particle_number(mol_data, docc_orb=None, act_orb=None, mapping="jordan_wigner"):
+    r"""Builds the particle number operator :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha` and
+    represent it in the basis of Pauli matrices.
```

Comment:
I like the new name!

However something sounds a bit off in the docstring summary. Should it be as follows?
```suggestion
    r"""Computes the particle number operator :math:`\hat{N}=\sum_\alpha \hat{n}_\alpha` of the
    molecule in the Pauli basis.
```